xml/System/Environment.xml

```diff
@@ -1639,16 +1617,18 @@ The following example creates environment variables for the <xref:System.Environ
         <ReturnType>System.OperatingSystem</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets an <see cref="T:System.OperatingSystem" /> object that contains the current platform identifier and version number.</summary>
-        <value>An object that contains the platform identifier and version number.</value>
+        <summary>Gets the current platform identifier and version number.</summary>
+        <value>The platform identifier and version number.</value>
         <remarks>
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks
 
-The `Environment.OSVersion` property does not provide a reliable way to identify the exact operating system and its version. Therefore, we do not recommend that you use this method. Instead: 
+In .NET 5.0 and later versions, the `Environment.OSVersion` property returns the actual version of the operating system. For more information, see [Environment.OSVersion returns the correct operating system version](/dotnet/core/compatibility/corefx#environmentosversion-returns-the-correct-operating-system-version).
+
+In .NET Framework and .NET Core 1.0 - 3.1, this property does *not* provide a reliable way to identify the exact operating system version. Instead of checking this property, consider one of these options:
 
-- To identify the operating system platform, use the <xref:System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform%2A?displayProperty=nameWithType> method. 
+- To identify the operating system platform, use the <xref:System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform%2A?displayProperty=nameWithType> method.
```

Comment:
This doesn't actually provide any version detection.  RuntimeInformation.IsOSPlatform can detect an OSPlatform which is just Windows, Linux, OSX, etc.  https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.osplatform?view=netcore-3.1

RuntimeInformation.OSDescription provided a reliable string with version (for logging purposes) but we don't recommend folks deconstruct this string.